### PR TITLE
Move particle plugin declarations to relative path instead of absolulte path.

### PR DESCRIPTION
**Pull request by MFraters**
Related to #5877. A first step for allowing multiple particle fields, by making the particle plugin path relative instead of absolute.

### CODE REVIEW

**Comment by gassmoeller** [comment]
Like discussed, I think it is not particularly beautiful, because the parameter parsing in the particle plugins now looks differently from all other plugins, but I think it is the only reasonable solution. Why do you only do this in the `parse_parameters` function and not in the `declare_parameters` function? I think they should at least look similar.

**Comment by MFraters** [comment]
I am doing it in both. I may have missed something? I am currently trying to fix the tests, which may show where I missed something. I will let you know as soon as I have fixed the tests and it is ready to review.

**Comment by MFraters** [comment]
This pull request is now ready for review. All testers passed, but I noticed some parts where not covered while debuging,so I also have added some new test :)

**Review by gassmoeller — CHANGES_REQUESTED** [review]
let me talk to you about an idea I had

**Review by tjhei — APPROVED** [review]

**Review by tjhei — COMMENTED** [review]
Looks good to go from my side. @gassmoeller ?

File: source/particle/property/interface.cc [hunk 1]
```diff
@@ -731,6 +720,8 @@ namespace aspect
 
         // lastly store internal integrator properties
         property_list.emplace_back (std::make_unique<IntegratorProperties<dim>>());
+
+        property_list.back()->declare_parameters (prm);
```
Comment (gassmoeller) [inline source/particle/property/interface.cc:724]: Why did this work before you added this line? Are the declare_parameters functions of the plugins called from somewhere else?

**Review by gassmoeller — COMMENTED** [review]
Yes, I think this is the best way. Can you check my one comment?

**Comment by MFraters** [comment]
@gassmoeller I have addressed your comments

**Review by gassmoeller — APPROVED** [review]
Looks great now. Thanks for testing my suggestions, even if they dont work out! :+1:

**Pull request merged**